0b.md

```diff
@@ -0,0 +1,273 @@
+- **inactive**: a sub account that has been inactivated (but not compromised)
+- **revoked**: a sub account that has been revoked (it may have been compromised, and all events posted on-behalf of master need to be invalidated)
+
+### On-Behalf of Attestations List
+
+This NIP introduces a replaceable event with `kind 10100`, which is used to keep a [NIP-51](https://github.com/nostr-protocol/nips/blob/master/51.md) list of `active`, `inactive` or `revoked` public keys, for publishing events on-behalf of a master account. The only valid tag in this list is the `p` tag as defined below.
+
+`.content` fields is unused and can be ignored if present.
+
+This list is owned by the master account, and can only be updated by an event signed by the master account itself, so never using a `b` on-behalf tag (see below), neither by [NIP-41](https://github.com/nostr-protocol/nips/blob/master/41.md) `p` tag in `kind 0`, nor using a [NIP-26](https://github.com/nostr-protocol/nips/blob/master/26.md) `delegation` tag.
+
+This list is ever growing, so each update has to include all previous attestations unchanged, and add one or more attestations to the list, otherwise they should be considered invalid and ignored by clients and relays.
+
+### Tag `b` (on-Behalf) for events
+
+This NIP introduces a new tag: `b`, which is indexable by relays, and can be present in active account events, if attested at the on-behalf list above, formatted as follows:
+
+```json
+[ "b", <pubkey of the master account> ]
+```
+
+To prevent spam, only one `b` tag can be present in an event.
+
+If present and valid, the `b` tag will represent that clients and relays should consider that event as published by the master account specified in the `b` tag, instead of the one in the event `.pubkey` field.
+
+If an invalid `b` tag is present, clients and relays should disregard the event, as if it had an invalid signature.
+
+### Tag `p` (with attestation string) for `kind 10100` list
+
+This NIP also introduces a new attestation string for `p` tag. The `p` tag can be present multiple times in master account's `kind 10100` On-Behalf Of attestations list, formatted as follows:
+
+```json
+[
+  "p", <pubkey>, <main relay URL>, <attestation string>
+]
+```
+
+Field `main relay URL` can be omitted if not required or needed for accessing the sub account events, and in this case an empty string should be used in the second field of the `p` tag.
+
+#### Active Attestation String
+
+The **Active Attestation** should be a string in the following format:
+
+```
+"active:<timestamp>:<kinds comma separated list, optional>"
+```
+
+Last `:` is only required if an optional kinds list is present. Without a kinds list, the attestation is valid for all event kinds besides `kind 10100`.
+
+#### Inactive and Revoked Attestation String
+
+The **Inactive** or **Revoked Attestations** should be a string in the following formats:
+
+```
+"inactive:<timestamp>"
+```
+
+or 
+
+```
+"revoked:<timestamp>"
```

Comment:
What's the purpose of the revoked timestamp if it is going to invalidate everything? 

It's easier to just delete the key from the list.

Reply:
The list has to keep all history to keep track of revocations, exactly to avoid replay attacks of past events